db_sql.go

```diff
@@ -648,28 +649,97 @@ func (x *sqlSessionDB) Read(sessionkey string) (*Token, error) {
 			return nil, err
 		}
 	} else {
-		if err := token.Permit.Deserialize(epermit); err != nil {
-			return nil, err
+		t, err2, done := x.populateToken(token, epermit, userId, oauthID)
+		if done {
+			return t, err2
 		} else {
-			token.UserId = UserId(userId)
-			user, err := getUserFromUserId(x.db, token.UserId)
-			if err != nil {
-				return nil, err
-			}
-			token.Username = user.Username
-			token.Email = user.Email
-			if user.Type == UserTypeLDAP {
-				token.Identity = user.Username
+			return nil, nil
+		}
+	}
+}
+
+func (x *sqlSessionDB) populateToken(token *Token, epermit string, userId int64, oauthID sql.NullString) (*Token, error, bool) {
```

Comment:
I might be wrong... but I don't think you need the `bool` at the end. At least, not with the code that I can see in this PR.

Reply:
Removed.